[JSC] Add more explicit exception checks to objectProtoFuncLookupGetter objectProtoFuncLookupSetter

Constellation opened the pull request:
#### 6e08c53f024309fbd2ab8dd72a14828d022b7803
<pre>
[JSC] Add more explicit exception checks to objectProtoFuncLookupGetter objectProtoFuncLookupSetter
<a href="https://bugs.webkit.org/show_bug.cgi?id=245304">https://bugs.webkit.org/show_bug.cgi?id=245304</a>
&lt;rdar://98498151&gt;

Reviewed by Mark Lam.

getOwnPropertyDescriptor call was missing explicit exception check before. It was ok because we are checking the result
value. But let&apos;s just deploy simpler way here and suppress failures in debug build.

* Source/JavaScriptCore/runtime/ObjectPrototype.cpp:
(JSC::JSC_DEFINE_HOST_FUNCTION):

Canonical link: <a href="https://commits.webkit.org/254581@main">https://commits.webkit.org/254581@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/fbf31cb6cfb92575fe878b8e3e54ede7cbc6f937

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/89519 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/34071 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/20240 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/98824 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/155131 "Built successfully and passed tests") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/32565 "Built successfully") | [  ~~🛠 mac-debug~~](https://ews-build.webkit.org/#/builders/71/builds/28047 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/81887 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/93235 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/95166 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/68/builds/25869 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/76397 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/25814 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/80747 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/80632 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/68806 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/81192 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/30323 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/14708 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/75015 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/30070 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/15643 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/26405 "Passed tests") | 
| [❌ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/3219 "Failed to push commit to Webkit repository") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/33518 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/38599 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/77883 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/32228 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/34734 "Passed tests") | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/17243 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

MenloDorian commented on Source/JavaScriptCore/runtime/ObjectPrototype.cpp:
```diff
@@ -225,7 +225,9 @@ JSC_DEFINE_HOST_FUNCTION(objectProtoFuncLookupGetter, (JSGlobalObject* globalObj
             ASSERT(slot.slotBase());
-            if (slot.slotBase()->getOwnPropertyDescriptor(globalObject, propertyName, descriptor))
+            bool result = slot.slotBase()->getOwnPropertyDescriptor(globalObject, propertyName, descriptor);
+            RETURN_IF_EXCEPTION(scope, { });
```
You can do this instead:
```
EXCEPTION_ASSERT(!scope.exception() || !result);
```

Constellation replied:
Changed.

MenloDorian commented on Source/JavaScriptCore/runtime/ObjectPrototype.cpp:
```diff
@@ -255,7 +257,9 @@ JSC_DEFINE_HOST_FUNCTION(objectProtoFuncLookupSetter, (JSGlobalObject* globalObj
             ASSERT(slot.slotBase());
-            if (slot.slotBase()->getOwnPropertyDescriptor(globalObject, propertyName, descriptor))
+            bool result = slot.slotBase()->getOwnPropertyDescriptor(globalObject, propertyName, descriptor);
+            RETURN_IF_EXCEPTION(scope, { });
```
ditto.

Constellation replied:
Changed.

MenloDorian commented on Source/JavaScriptCore/runtime/ObjectPrototype.cpp:
```diff
@@ -239,14 +241,14 @@ JSC_DEFINE_HOST_FUNCTION(objectProtoFuncLookupSetter, (JSGlobalObject* globalObj
     PropertySlot slot(thisObject, PropertySlot::InternalMethodType::GetOwnProperty);
     bool hasProperty = thisObject->getPropertySlot(globalObject, propertyName, slot);
-    EXCEPTION_ASSERT(!scope.exception() || !hasProperty);
+    RETURN_IF_EXCEPTION(scope, { });
```
Why this change?

Constellation replied:
Ditto.

MenloDorian commented on Source/JavaScriptCore/runtime/ObjectPrototype.cpp:
```diff
@@ -209,14 +209,14 @@ JSC_DEFINE_HOST_FUNCTION(objectProtoFuncLookupGetter, (JSGlobalObject* globalObj
     PropertySlot slot(thisObject, PropertySlot::InternalMethodType::GetOwnProperty);
     bool hasProperty = thisObject->getPropertySlot(globalObject, propertyName, slot);
-    EXCEPTION_ASSERT(!scope.exception() || !hasProperty);
+    RETURN_IF_EXCEPTION(scope, { });
```
Why this change?

Constellation replied:
Just for simplification, but reverted.

MenloDorian reviewed (approved):
r=me

Constellation committed:
[JSC] Add more explicit exception checks to objectProtoFuncLookupGetter objectProtoFuncLookupSetter
https://bugs.webkit.org/show_bug.cgi?id=245304
<rdar://98498151>

Reviewed by Mark Lam.

getOwnPropertyDescriptor call was missing explicit exception check before. It was ok because we are checking the result
value. But let's just deploy simpler way here and suppress failures in debug build.

* Source/JavaScriptCore/runtime/ObjectPrototype.cpp:
(JSC::JSC_DEFINE_HOST_FUNCTION):

Canonical link: https://commits.webkit.org/254581@main
